app/celer-g4/DetectorConstruction.cc

```diff
@@ -97,6 +107,43 @@ G4VPhysicalVolume* DetectorConstruction::Construct()
         sd.enabled = false;
     }
 
+    // Setup options for the magnetic field
+    auto field_type = GlobalSetup::Instance()->GetFieldType();
+    if (field_type == "rzmap")
+    {
+        auto map_filename = GlobalSetup::Instance()->GetFieldFile();
+        if (map_filename.empty())
+        {
+            G4Exception("DetectorConstruction::Construct()",
+                        "",
+                        FatalException,
+                        "No field file was specified with /celerg4/fieldFile");
+        }
+        CELER_LOG_LOCAL(info) << "Using RZMapField with " << map_filename;
+        use_field_map_ = true;
+
+        // Create celeritas::RZMapFieldParams from input
+        RZMapFieldInput rz_map;
+        std::ifstream(map_filename) >> rz_map;
+        field_params_ = std::make_shared<RZMapFieldParams>(rz_map);
+
+        celeritas::app::GlobalSetup::Instance()->GetAlongStepOptions()
+            = celeritas::RZMapFieldAlongStepFactory([=] { return rz_map; });
```

Comment:
We should add a factory function that takes RZ map field params rather than the map input: as implemented, we end up with two copies of the field params both on host and device.

Reply:
Actually I try to pass RZMapFieldParams  when I added RZMapFieldAlongStepFactory, but failed.  So, I need your help on this or I will try again in a separate MR as the change also need testing within CMSSW.